src/Widgets/SettingsPopover.vala

```diff
@@ -0,0 +1,204 @@
+        dark_button.tooltip_text = _("Dark");
+        dark_button.group = hc_button;
+
+        Gtk.CssProvider custom_button_provider;
+
+        var custom_button = add_theme_button (Themes.CUSTOM, out custom_button_provider);
+        custom_button.tooltip_text = _("Custom");
+        custom_button.group = hc_button;
+        custom_button.get_style_context ().add_class ("color-custom");
+
+        var natural_copy_paste_button = new Granite.SwitchModelButton (_("Natural Copy/Paste")) {
+            description = _("Shortcuts don’t require Shift; may interfere with CLI apps"),
+            active = settings_actions.get_action_state ("natural-copy-paste").get_boolean (),
+            action_name = ACTION_NATURAL_COPY
+        };
+
+        var box = new Gtk.Box (Gtk.Orientation.VERTICAL, 6) {
+            margin_bottom = 6,
+            margin_top = 12,
+        };
+
+        box.add (font_size_box);
+        box.add (new Gtk.Separator (Gtk.Orientation.HORIZONTAL));
+        box.add (theme_box);
+        box.add (new Gtk.Separator (Gtk.Orientation.HORIZONTAL));
+        box.add (natural_copy_paste_button);
+        box.show_all ();
+        child = box;
+
+        switch (settings_actions.get_action_state ("theme").get_string ()) {
+            case Themes.HIGH_CONTRAST:
+                hc_button.active = true;
+                break;
+
+            case Themes.LIGHT:
+                light_button.active = true;
+                break;
+
+            case Themes.CUSTOM:
+                custom_button.active = true;
+                break;
+
+            case Themes.DARK:
+            default:
+                dark_button.active = true;
+                break;
+        }
+
+        custom_button.clicked.connect (() => {
+            if (custom_button.active) {
+                show_theme_editor ();
+                popdown ();
+            }
+        });
+
+        Application.settings.changed["background"].connect (() => {
+            update_theme_provider (custom_button_provider, Themes.CUSTOM);
+        });
+
+        Application.settings.changed["foreground"].connect (() => {
+            update_theme_provider (custom_button_provider, Themes.CUSTOM);
+        });
+
+        settings_actions.action_state_changed.connect ((n, s) => {
+            if (n == "zoom") {
+                zoom_default_button.label = "%.0f%%".printf (s.get_double () * 100);
+            }
+        });
+
+        follow_system_button.bind_property (
+            "active",
+            theme_revealer,
+            "reveal-child",
+            BindingFlags.SYNC_CREATE | BindingFlags.INVERT_BOOLEAN
+        );
+    }
+
+    private Gtk.RadioButton add_theme_button (string theme, out Gtk.CssProvider css_provider = null) {
+        var button = new Gtk.RadioButton (null) {
+            action_name = ACTION_THEME,
+            action_target = new Variant.string (theme),
+            halign = Gtk.Align.CENTER
+        };
+
+        if (css_provider == null) {
```

Comment:
This gives two new compile warnings with valac 0.56.7 
```
../src/Widgets/SettingsPopover.vala:180.13-180.32: warning: Use of possibly unassigned parameter `css_provider'
  180 |         if (css_provider == null) {
      |             ^~~~~~~~~~~~~~~~~~~~   
../src/Widgets/SettingsPopover.vala:186.9-186.90: warning: Use of possibly unassigned parameter `css_provider'
  186 |         style_context.add_provider (css_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);

```
Do you know if later versions of Vala will support it? I tendto avoid this pattern though it is useful.